core/src/test/java/com/yahoo/tagchowder/ParserTest.java

```diff
@@ -51,6 +52,69 @@ public void testSampleHtml() throws IOException, SAXException {
         parser.parse(inSource);
     }
 
+    /**
+     * Parse an sample AMP html.
+     *
+     * @throws IOException IOException
+     * @throws SAXException SAXException
+     */
+    @Test
+    public void testSampleAMPHtml() throws IOException, SAXException {
+        final String html = getSampleHtml("amphtml.txt");
+        final Parser parser = new Parser();
+        parser.setFeature(Parser.AMP_VALIDATION_FEATURE, true);
+        final InputSource inSource = new InputSource(new StringReader(html));
+        parser.parse(inSource);
```

Comment:
Other UTs were created plainly for coverage or just want to make sure parse() wont throw exception.
I think your use case is more into the functionality part.

Can you create your own handler which extends DefaultHandler (override startElement()) and do something like following code? Then can you make sure you are getting the correct element you want. Just give an idea here.
We need to make sure the functionality you need for AMP is working, making sure parse() function works is just not enough.

`final CustomHandler consumer = new CustomHandler(...);

final Parser p = new Parser();
p.setContentHandler(consumer);
p.setErrorHandler(consumer);
p.setFeature(p.AMP_VALIDATION_FEATURE, true);
p.setDefaultBufferSize(html.length() + ADDITIONAL_BUFF_SIZE);
p.parse(new InputSource(new StringReader(html)));

Assert.assertTrue(consumer.numberOf4email(), 1);
....
`

Reply:
Where is the UT for the Parser in the past when it was modified?

Reply:
there was no UT at all for this Parser. When we forked from tagsoup, we need one simpliest UT: ParserTest.

Reply:
This covers in the AMP validator itself.

Reply:
yes, I understand that. The new code is added into tagchowder which will live by its own. People may start "copy" code and assume this code part is unbeatable while no UT is there to 100% verify. 

You can port some simple one here, if that is easier.

Reply:
Done